src/radical/pilot/agent/executing/popen.py

```diff
@@ -474,5 +559,236 @@ def _check_running(self):
         return action
 
 
+    # --------------------------------------------------------------------------
+    def _get_check(self, event):
+
+        return '\\\n        || (echo "%s failed"; false) || exit 1\n' % event
+
+
+    # --------------------------------------------------------------------------
+    #
+    # launcher
+    #
+    def _get_prof(self, event, tid, msg=''):
+
+        return '$RP_PROF %s\n' % event
+
+
+    # --------------------------------------------------------------------------
+    #
+    # launcher
+    #
+    def _get_launch_env(self, task, launcher):
+
+        ret  = ''
+        cmds = launcher.get_launcher_env()
+        for cmd in cmds:
+            ret += '%s %s' % (cmd, self._get_check('launcher env'))
+        return ret
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _get_pre_launch(self, task, launcher):
+
+        ret  = ''
+        cmds = task['description']['pre_launch']
+        for cmd in cmds:
+            ret += '%s %s' % (cmd, self._get_check('pre_launch failed'))
+
+        return ret
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _get_launch_cmds(self, task, launcher, exec_path):
+
+
+        ret  = ''
+        cmds = ru.as_list(launcher.get_launch_cmds(task, exec_path))
+        for cmd in cmds:
+            ret += '%s %s' % (cmd,  self._get_check('launch failed'))
+
+        return ret
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _get_post_launch(self, task, launcher):
+
+        ret  = ''
+        cmds = task['description']['post_launch']
+        for cmd in cmds:
+            ret += '%s %s' % (cmd, self._get_check('post_launch failed'))
+
+        return ret
+
+
+    # --------------------------------------------------------------------------
+    # exec
+    #
+    def _get_rp_env(self, task):
+
+        tid  = task['uid']
+        name = task['name'] or tid
+        sbox = os.path.realpath(task['task_sandbox_path'])
+
+        if sbox.startswith(self._pwd):
+            sbox = '$RP_PILOT_SANDBOX%s' % sbox[len(self._pwd):]
+
+        ret  = ''
+        ret += 'export RP_TASK_ID="%s"\n'          % tid
+        ret += 'export RP_TASK_NAME="%s"\n'        % name
+        ret += 'export RP_PILOT_ID="%s"\n'         % self._cfg['pid']
+        ret += 'export RP_SESSION_ID="%s"\n'       % self.sid
+        ret += 'export RP_RESOURCE="%s"\n'         % self.resource
+        ret += 'export RP_RESOURCE_SANDBOX="%s"\n' % self.rsbox
+        ret += 'export RP_SESSION_SANDBOX="%s"\n'  % self.ssbox
+        ret += 'export RP_PILOT_SANDBOX="%s"\n'    % self.psbox
+        ret += 'export RP_TASK_SANDBOX="%s"\n'     % sbox
+        ret += 'export RP_LFS="%s"\n'              % self.lfs
+        ret += 'export RP_GTOD="$RP_PILOT_SANDBOX/gtod"\n'
+
+        if self._prof.enabled:
+            ret += 'export RP_PROF="$RP_PILOT_SANDBOX/prof"\n'
```

Comment:
`RP_GTOD` and `RP_PROF` are set as attributes in `AgentExecutingComponent` base class, should they be deleted from there? (I would prefer to delete them from there, otherwise they should be referenced here, e.g., `ret += 'export RP_GTOD="%s"\n' % self.gtod`)

Reply:
Thanks, I cleaned that up (fixed here)